packages/block-editor/src/hooks/layout.js

```diff
@@ -180,11 +177,9 @@ function LayoutPanelPure( {
 	}
 
 	// Only show the inherit toggle if it's supported,
-	// a default theme layout is set (e.g. one that provides `contentSize` and/or `wideSize` values),
 	// and either the default / flow or the constrained layout type is in use, as the toggle switches from one to the other.
 	const showInheritToggle = !! (
 		allowInheriting &&
-		!! defaultThemeLayout &&
```

Comment:
So I don't really know if this is safe. 

Things appear to work as expected.

The scenario is that:

- `allowInheriting` is `true`, but
- `defaultThemeLayout` has no value.

A value of `{}` would pass the test, so it seems safe to just wave it through, unless we specifically want to check for `settings.layout.wideSize|contentSize`? 

Not sure 🤔

Reply:
Good question! I haven't taken this PR for a spin, but does this affect classic themes at all? And is it a problem if when you toggle the inherit toggle, if it doesn't immediately apply any rules? I.e. if folks go to set child blocks to "wide" versus content alignment they mightn't see any difference until a value is entered in for the constrained controls 🤔

These are just drive-by comments, though, thanks for working on the fix! If it's still open in a couple of days, happy to take it for a spin and give it a proper review 🙂

Reply:
> I haven't taken this PR for a spin, but does this affect classic themes at all? 

Good question. I smoke tested a couple and couldn't see any difference between this PR and trunk, but I didn't really go in depth. 

I'm just wondering if the approach is any good right now 🤷🏻 

> And is it a problem if when you toggle the inherit toggle, if it doesn't immediately apply any rules? I.e. if folks go to set child blocks to "wide" versus content alignment they mightn't see any difference until a value is entered in for the constrained controls

I'm not sure - I can't see any issue, but I'm a bit layout blind. 

Also, the fact that, on trunk,`settings.layout` can be an empty object and pass the test is a bit suspect. 

If we tighten up that test, so that we need `settings.layout` to actually contain something, then we'd have to work out how to deal with the empty layout panel.

Whatever folks think is the best way forward...

Reply:
Given that this whole UI is going to change once #53455 (or something similar) is ready, it's not too important to make this a perfect fix. 

> is it a problem if when you toggle the inherit toggle, if it doesn't immediately apply any rules?

I think as long as the controls to set custom content/wide withs are shown, it's not super awkward 😅 because it exposes those controls, which once set will produce the visible effect.

We could perhaps refine the check to make sure that _either_ the theme sets a `contentSize` or `allowCustomContentAndWideSize` is true, because if neither exist we might as well hide the whole panel.